### DISCUSSION TITLE
Creating Quarto documents programmatically

### DISCUSSION THREAD

**Discussion by lorenzoFabbri** [quarto-dev/quarto-cli#5373] [Q&A] [answered]
I would like to create Quarto documents programmatically:

1. I call the function `create_document` passing as input just the path to a yaml file containing the DOI (a alphanumeric string) and a DAG (which is just a string separated by new lines here).
2. This functions should ideally create a new Quarto document in a given directory.
3. The Quarto document should be parametrised (both YAML and content) using information resulting from the call to `create_document`.

The YAML currently looks like this:
```
---
title: "Some title"
date: 09/01/2014
doi: doi_from_file
---
```

The body of the document has the following structure:
```
## Important links

- [Article](https://doi.org/{{< meta doi >}})

## DAG
```

```{r}
dag <- dagitty::dagitty(dag_from_file)
tidy_dag <- ggdag::tidy_dagitty(dag)

ggdag::ggdag_status(tidy_dag) +
  ggdag::theme_dag()
```

It is important to note that `doi_from_file` and `dag_from_file` result from the call to `create_document`. Is there a way to achieve this?

**Comment by mcanouil** [comment, marked as answer]
Quarto documents are simple flat text files.

Simply write the file with whatever you like in R.

````r
cat(sprintf(
'---
title: "Some title"
date: 09/01/2014
doi: %s
---

## Important links

- [Article](https://doi.org/{{< meta doi >}})

## DAG

```{r}
dag <- dagitty::dagitty(%s)
tidy_dag <- ggdag::tidy_dagitty(dag)

ggdag::ggdag_status(tidy_dag) +
  ggdag::theme_dag()
```
',
doi_from_file,
dag_from_file
), file = 'index.qmd')
````

```r
quarto::quarto_render("index.qmd")
```

Or create a template document that you can reuse (see <https://quarto.org/docs/computations/parameters.html>):
````
---
params:
  doi: none
  dag: none
title: "Some title"
date: 09/01/2014
doi: '{{< meta params.doi >}}'
---

## Important links

- [Article](https://doi.org/{{< meta doi >}})

## DAG

```{r}
if (!params[["dag"]] %in% "none") {
  dag <- dagitty::dagitty(params[["dag"]])
  tidy_dag <- ggdag::tidy_dagitty(dag)

  ggdag::ggdag_status(tidy_dag) +
    ggdag::theme_dag()
}
```
````

**Reply by mcanouil** [reply]
Additionally, see <https://quarto-dev.github.io/quarto-r/> for rendering from R using the R wrapper package.
